src/app/modules/twilio/twilio.controller.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusCodes } from 'http-status-codes'
+import Twilio from 'twilio'
+
+import { ITwilioSmsWebhookBody } from 'src/types/twilio'
+
+import { ControllerHandler } from '../core/core.types'
+
+import { logFailedSmsDelivery } from './twilio.service'
+
+const smsDeliveryFailedStatus = ['undelivered', 'failed']
+
+/**
+ * Middleware which validates that a request came from Twilio Webhook
+ */
+const validateTwilioWebhook = Twilio.webhook()
+
+/**
+ * Process the Webhook requests if they are failed or unsuccessful, ignoring the rest
+ *
+ * @param req Express request object
+ * @param res - Express response object
+ */
+export const twilioSmsUpdates: ControllerHandler<
+  unknown,
+  never,
+  ITwilioSmsWebhookBody
+> = async (req, res) => {
+  if (smsDeliveryFailedStatus.includes(req.body.MessageStatus))
```

Comment:
should we also log every other message status, using `logger.info` instead of `logger.error`?

Reply:
If this is the case, should I remove twilio.service and have logging of both error and non-errors on the controller level?